app/controllers/application_controller.rb

```diff
@@ -32,6 +32,10 @@ def path_based_on_role
     end
   end
 
+  def user_params
+    params.require(:user).permit(:username, :password, :password_confirmation, :gender)
+  end
```

Comment:
Good job recognizing that this should be moved into application controller helper